src/test/java/seedu/address/testutil/BookingBuilder.java

```diff
@@ -0,0 +1,66 @@
+package seedu.address.testutil;
+
+import java.time.LocalDate;
+
+import seedu.address.model.person.Guest;
+import seedu.address.model.room.booking.Booking;
+import seedu.address.model.room.booking.BookingPeriod;
+
+/**
+ * A utility class to help with building Room objects.
+ */
+public class BookingBuilder {
+
+    public static final Guest DEFAULT_GUEST = new GuestBuilder().build();
+    public static final BookingPeriod DEFAULT_BOOKING_PERIOD =
+        new BookingPeriod(LocalDate.now().format(BookingPeriod.FORMAT),
+            LocalDate.now().plusDays(1).format(BookingPeriod.FORMAT));
+    public static final Boolean DEFAULT_CHECKIN = false;
+
+    private Guest guest;
+    private BookingPeriod bookingPeriod;
+    private Boolean checkIn;
+
+    public BookingBuilder() {
+        guest = DEFAULT_GUEST;
+        bookingPeriod = DEFAULT_BOOKING_PERIOD;
+        checkIn = DEFAULT_CHECKIN;
+    }
+
+    /**
+     * Initializes the RoomBuilder with the data of {@code roomToCopy}.
+     */
+    public BookingBuilder(Booking bookingToCopy) {
+        guest = bookingToCopy.getGuest();
+        bookingPeriod = bookingToCopy.getBookingPeriod();
+        checkIn = bookingToCopy.isCheckedIn();
+    }
+
+    /**
+     * Sets the {@code Name} of the {@code Room} that we are building.
+     */
+    public BookingBuilder withGuest(Guest guest) {
+        this.guest = new Guest(guest);
```

Comment:
What's the benefit of this over `this.guest = guest`? I understand it's also used in the original AB4 to build `Person`s. 

Applies to both `BookingBuilder` and `RoomBuilder`.

Reply:
Creates a new object, instead of just pointing to the referenced object. It's needed for testing where we make copies of an object to test Before/After states.